### PR TITLE
Attempt to discover a cross-compile toolchain version of ar, just like objcopy

### PR DESCRIPTION
**Pull request by jnicholls**
This pattern may warrant something hoisted into the gcc crate, in my opinion.  This is becoming very common-place.

### CODE REVIEW

File: backtrace-sys/build.rs [hunk 1]
```diff
@@ -72,11 +84,10 @@ fn main() {
     let tmpdir = dst.join("__tmp");
     drop(fs::remove_dir_all(&tmpdir));
     t!(fs::create_dir_all(&tmpdir));
-    run(Command::new("ar").arg("x").arg(&lib).current_dir(&tmpdir), "ar");
+    run(Command::new(&ar).arg("x").arg(&lib).current_dir(&tmpdir), "ar");
```
Comment (alexcrichton) [inline backtrace-sys/build.rs]: Could you change "ar" here at the end to &ar as well?

File: backtrace-sys/build.rs [hunk 2]
```diff
@@ -97,7 +108,7 @@ fn main() {
         objs.push(obj.path());
     }
 
-    run(Command::new("ar").arg("crus").arg(&lib).args(&objs), "ar");
+    run(Command::new(&ar).arg("crus").arg(&lib).args(&objs), "ar");
```
Comment (alexcrichton) [inline backtrace-sys/build.rs]: (same as above)

**Comment by alexcrichton** [comment]
Looks great, thanks!

**Comment by jnicholls** [comment]
Updated the commit per your request.  Thanks.

**Pull request merged**